### PR TITLE
WIP: transform-next

### PR DESCRIPTION
**Pull request by johan-bjareholt**
- Removed ChunkedEvents resource
- Removed View resources
- Added Query resource based on query2 with built-in caching

### CODE REVIEW

**Comment by johan-bjareholt** [comment]
Build fails because aw-core is not updated

File: aw_server/rest.py [hunk 1]
```diff
@@ -177,35 +179,18 @@ def post(self, bucket_id):
...
-    @api.param("end", "End datetime of events to query over")
-    @copy_doc(ServerAPI.query_view)
-    def get(self, viewname):
-        args = request.args
-        start = iso8601.parse_date(args["start"]) if "start" in args else None
-        end = iso8601.parse_date(args["end"]) if "end" in args else None
+# QUERY
 
-        result = app.api.query_view(viewname, start, end)
+# It's not possible to expect a raw string in restplus/swagger, stupid
```
Comment (ErikBjare) [inline aw_server/rest.py]: I think the way to do this is to simply create a query model that contains only one field. Something like this:

```py
query_fields = api.model('Query', {
    'query': fields.String,
})
```

Is there any reason to not do it like this?

**Comment by johan-bjareholt** [comment]
I did that initially actually, then undid it due to it was annoying to do
multi-line strings and had to do escaping multiple times.

**Comment by ErikBjare** [comment]
We might want to consider using semicolons instead of newlines in the syntax. Kind of makes more sense in a DSL like this in my opinion (less fiddling with newlines).

I'd still prefer to have the JSON query model there. A clean REST API should rarely - if ever - have the payloads delivered in non-JSON format. I don't mind wrapping the DSL code in a JSON object.

Perhaps some parts of the DSL language (such as the start/end dates) would be better specified as arguments instead of as a part of the DSL? I might be wrong about that though. Regardless I think it's nice to have the JSON wrapper for the sake of consistency.

**Comment by johan-bjareholt** [comment]
It's just so incredibly clunky to do multi-line JSON strings, so to write it in in a readable manner in python was a hassle. Regarding starttime, endtime and name I agree though. Can look into it again, but since JSON technically does not support multi-line strings we will have to do workarounds no matter what language it is, so it might get very ugly, but as I said I'll look into it. Readability counts.

**Comment by ErikBjare** [comment]
@johan-bjareholt In the future, please don't remove code as soon as a replacement is implemented. Instead, comment or otherwise mark it as deprecated.

That way we won't have to ensure that all dependents on the deprecated code are migrated right away. If you, for example, would have kept the old views endpoint here then we could have merged this a lot earlier because we wouldn't have to migrate all the code using views to query2 before merging.

**Pull request merged**